lib/linkerd2-trace-context/src/lib.rs

```diff
@@ -0,0 +1,100 @@
+use bytes::Bytes;
+use futures::Sink;
+use linkerd2_error::Error;
+use rand::Rng;
+use std::fmt;
+use std::time::SystemTime;
+
+pub mod layer;
+mod propagation;
+
+pub use layer::layer;
+
+const SPAN_ID_LEN: usize = 8;
+
+#[derive(Debug, Default)]
+pub struct Id(Vec<u8>);
```

Comment:
Correct me if I'm wrong, but this is only ever either an 8-byte span ID or a 16-byte trace ID, right? Why not have separate types for these, like
```rust
#[derive(Debug, Default)]
pub struct SpanId([u8; 8]);

#[derive(Debug, Default)]
pub struct TraceId([u8; 16]);
```
That way, we wouldn't have to allocate a `Vec` on the heap to represent each ID.

Alternatively, we could do
```rust
#[derive(Debug, Default)]
pub struct SpanId(u64);

#[derive(Debug, Default)]
pub struct TraceId(u128);
```
if we don't want to index the individual bytes.

Reply:
I really like these suggestions! I don't want them to block merging this, though. Would you mind opening a p0 in the proxy backlog to remove the Vecs from the trace-context code, @hawkw ?

Reply:
We use Vec throughout for Ids because `hex::decode` returns a Vec and the Span exporter gRPC client requires Vecs.  We could convert these to arrays in the meantime, but I don't think it saves on allocations.

Reply:
@adleong oh, that makes sense, never mind...there's been some talk in the past about patching `prost` to allow `Bytes` to be used as protobuf `bytes` fields; if we did that, we could probably just split the bytes we're parsing from into these types...

Reply:
the version suggestion seems like a fine followup, though probably not high priority